v0/destinations/webengage/transform.js

```diff
@@ -0,0 +1,143 @@
+const { EventType } = require("../../../constants");
+const {
+  CONFIG_CATEGORIES,
+  MAPPING_CONFIG,
+  WEBENGAGE_IDENTIFY_EXCLUSION,
+  FINAL_ENDPOINT
+} = require("./config");
+
+const {
+  defaultRequestConfig,
+  constructPayload,
+  defaultPostRequestConfig,
+  CustomError,
+  ErrorMessage,
+  extractCustomFields
+} = require("../../util");
+
+const responseBuilder = (message, category, { Config }) => {
+  let payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  let baseUrl, endPoint;
+  const { dataCenter, licenseCode, apiKey } = Config;
+  if (!payload) {
+    // fail-safety for developer error
+    throw new CustomError(ErrorMessage.FailedToConstructPayload, 400);
+  }
+  if (!payload.userId && !payload.anonymousId) {
+    throw new CustomError(
+      "[WEBENGAGE]: Either one of userId or anonymousId is mandatory.",
+      400
+    );
+  }
+
+  if (category.type === "identify") {
+    const eventTimeStamp = payload.birthDate;
+    if (eventTimeStamp === "Invalid date") {
+      throw new CustomError(
+        "[WEBENGAGE]: birthday must be in this (YYYY-MM-DD) format.",
+        400
+      );
+    }
+    const customAttributes = {};
+    payload = {
+      ...payload,
+      attributes: extractCustomFields(
+        message,
+        customAttributes,
+        ["context.traits", "traits"],
+        WEBENGAGE_IDENTIFY_EXCLUSION
+      )
+    };
+    endPoint = `${FINAL_ENDPOINT(dataCenter)}/${licenseCode}/users`;
+  } else {
+    const eventTimeStamp = payload.eventTime;
+    if (eventTimeStamp === "Invalid date") {
+      throw new CustomError(
+        "[WEBENGAGE]: timestamp must be ISO format (YYYY-MM-DDTHH:mm:ss.sssZ).",
+        400
+      );
+    }
+    endPoint = `${FINAL_ENDPOINT(dataCenter)}/${licenseCode}/events`;
+  }
+  const response = defaultRequestConfig();
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.headers = {
+    "Content-Type": "application/json",
+    Authorization: `Bearer ${apiKey}`
+  };
+  response.endpoint = endPoint;
+  response.body.JSON = payload;
+  return response;
+};
+
+const processEvent = (message, destination) => {
+  if (!message.type) {
+    throw Error("[WEBENGAGE]: Message Type is not present. Aborting message.");
+  }
+  const messageType = message.type.toLowerCase();
+  switch (messageType) {
+    case EventType.IDENTIFY:
+      return responseBuilder(message, CONFIG_CATEGORIES.IDENTIFY, destination);
+    case EventType.TRACK:
+      return responseBuilder(message, CONFIG_CATEGORIES.EVENT, destination);
+    case EventType.PAGE:
+    case EventType.SCREEN:
+      category = CONFIG_CATEGORIES.EVENT;
+      const name = message.name
+        ? ` ${message.name}`
+        : message.properties?.name
+        ? ` ${message.properties.name}`
+        : "";
+      const categoryName = message.properties.category
+        ? ` ${message.properties.category}`
+        : "";
+      message.event = `Viewed${name}${categoryName} ${messageType}`;
```

Comment:
Can we remove ternary and use if else (https://medium.com/@benlmsc/stop-using-nested-ternary-operators-heres-why-53e7e078e65a)